File: tests/csharp/tests-csharp-xunit-allure3-playwright/Tests/E2e/SessionTests.cs
using Allure.Net.Commons;
using Allure.Net.Commons.Attributes;
using Tests;

namespace Tests.E2e;

[AllureLabel("layer", "e2e")]
[AllureEpic("Authentication")]
[AllureFeature("Session")]
[AllureSeverity(SeverityLevel.critical)]
[AllureSuite("Session")]
public sealed class SessionTests : TestBase
{
    [Trait("TestCategory", "e2e")]
    [Trait("TestCategory", "negative")]
    [Fact(DisplayName = "Invalid token clears session and hides welcome")]
    public void InvalidTokenClearsSession()
    {
        HomePage.OpenPageWithInvalidToken()
            .ShouldHideWelcomePanel()
            .ShouldClearAuthToken();
    }

    [Trait("TestCategory", "e2e")]
    [Trait("TestCategory", "positive")]
    [Fact(DisplayName = "Session survives a page reload (token in localStorage)")]
    public void SessionSurvivesReload()
    {
        HomePage.OpenPageWithLocalStorageAuthentication("user1", "password1")
            .ShouldHaveWelcomeMessage("Welcome, user1!")
            .ReloadPage()
            .ShouldHaveWelcomeMessage("Welcome, user1!");
    }
}

File: tests/rust/tests-rust-testing-selenium/tests/e2e/session_tests.rs
use allure_cargotest::allure_test;
use tests_rust_testing_selenium as tests;

fn layer() {
    tests::layer_e2e("Session", "Authentication", "Session", "critical");
}

#[allure_test(name = "Invalid token clears session and hides welcome")]
#[tokio::test]
async fn invalid_token_clears_session() {
    layer();
    allure_rust_commons::tag("negative");
    tests::with_browser(|| async {
        tests::HomePage::default()
            .open_page_with_invalid_token()
            .await
            .should_hide_welcome_panel()
            .await
            .should_clear_auth_token()
            .await;
    })
    .await;
}

#[allure_test(name = "Session survives a page reload (token in localStorage)")]
#[tokio::test]
async fn session_survives_reload() {
    layer();
    allure_rust_commons::tag("positive");
    tests::with_browser(|| async {
        tests::HomePage::default()
            .open_page_with_local_storage_authentication("user1", "password1")
            .await
            .should_have_welcome_message("Welcome, user1!")
            .await
            .reload_page()
            .await
            .should_have_welcome_message("Welcome, user1!")
            .await;
    })
    .await;
}
